src/realm/object_id.hpp

```diff
@@ -83,10 +91,11 @@ class ObjectId {
 
     Timestamp get_timestamp() const;
     std::string to_string() const;
+    ObjectIdBytes to_bytes() const;
     size_t hash() const noexcept;
 
 private:
-    uint8_t m_bytes[12] = {};
+    ObjectIdBytes m_bytes = {};
```

Comment:
We rely on `sizeof(ObjectId)` in the ArrayObjectId implementation, so can you add a safeguard like the following somewhere `static_assert(sizeof(ObjectId) == 12, "changing the size of an ObjectId is a file format breaking change");` Just to make me feel better about swapping out the underlying implementation in case we run into a buggy implementation of std::array somewhere.

Reply:
Done.